src/utils.cpp

```diff
@@ -397,20 +397,22 @@ std::string resolve_binary_path(const std::string& cmd)
   }
 }
 
-void cat_file(const char *filename)
+void cat_file(const char *filename, size_t max_bytes)
 {
   std::ifstream file(filename);
-  std::string line;
 
   if (file.fail())
   {
     std::cerr << "ERROR: file not found: " << filename << std::endl;
     return;
   }
-  while (getline(file, line))
-  {
-    std::cout << line << std::endl;
+
+  std::string buf(max_bytes, '\0');
+  file.read(&buf[0], max_bytes);
```

Comment:
Could you add some comments around this code explaining it please? I see you're resizing the string later, but it's not that obvious and we don't want people modifying this and causing memory problems. Also, is the `\0` initialisation required?

Reply:
I think the `\0` is fine. The alternative is you do a `buf.reserve(max_bytes)`. But yeah comments might be nice. I had to scratch my head a little bit reading it.

Reply:
It's just that this code is stuff that is run during execution of a bpf program, so we'd ideally like it to be somewhat optimised - don't want to zero initialise 10k bytes if we don't have to

Reply:
Could even just use a char array and `puts()`

Reply:
The original implementation just reads until the first `\n` and prints that out. That includes any `\0` null bytes in the file. I wanted to stick to that behaviour as I'd expect `cat` to be a 1:1 copy of the file, although I do not have a use case for it. 
Browsing the C++ stdlib, `read` seemed to be the most suitable (and likely fastest, as it doesn't scan for a delim, it just copies). However as it reads into a `char *` the string has to be pre allocated to prevent a segv. I've tested it with `reserve` but couldn't get it to work.
To make sure we do not output extra nullbytes the string has to be resized if an EOF was hit, if only 1 bit was read  511 null bytes would still be printed.
Null bytes are also the reason why I want c++ strings instead of char arrays. With a char array it stops printing at the first null byte, however looking at it again using `cout.write(x, size)` might also be an option. I'd have to test that, sounds like a good option if it works.

I'll do a bit more testing. A loop construct to prevent really big buffers in case the user sets a huge size might also be useful too.

> Could even just use a char array and puts

I want to avoid `puts` to make #605 possible

Reply:
Good point about using C++ strings/not using puts. For the other issue, I think `buf.resize(xxx)` is the function you want, not `buf.reserve(xxx)`

Reply:
But I've looked it up and it does initialise all those characters... so that's no better than what you're already doing